framework/src/controller/schema/application.js

```diff
@@ -13,6 +13,211 @@ module.exports = {
 	constants: {
 		id: '#/app/constants',
 		type: 'object',
+		required: [
+			'ACTIVE_DELEGATES',
+			'BLOCK_SLOT_WINDOW',
+			'ADDITIONAL_DATA',
+			'BLOCK_RECEIPT_TIMEOUT',
+			'EPOCH_TIME',
+			'FEES',
+			'MAX_PAYLOAD_LENGTH',
+			'MAX_PEERS',
+			'MAX_SHARED_TRANSACTIONS',
+			'MAX_TRANSACTIONS_PER_BLOCK',
+			'MAX_VOTES_PER_TRANSACTION',
+			'MAX_VOTES_PER_ACCOUNT',
+			'MIN_BROADHASH_CONSENSUS',
+			'MULTISIG_CONSTRAINTS',
+			'NETHASHES',
+			'NORMALIZER',
+			'REWARDS',
+			'TOTAL_AMOUNT',
+			'UNCONFIRMED_TRANSACTION_TIMEOUT',
+			'EXPIRY_INTERVAL',
+		],
+		properties: {
+			ACTIVE_DELEGATES: {
+				type: 'integer',
+				min: 0,
+			},
+			BLOCK_SLOT_WINDOW: {
+				type: 'integer',
+			},
+			ADDITIONAL_DATA: {
+				type: 'object',
+				required: ['MIN_LENGTH', 'MAX_LENGTH'],
+				properties: {
+					MIN_LENGTH: {
+						type: 'integer',
+					},
+					MAX_LENGTH: {
+						type: 'integer',
+					},
+				},
+			},
+			BLOCK_RECEIPT_TIMEOUT: {
+				type: 'integer',
+			},
+			EPOCH_TIME: {
+				type: 'object',
+			},
+			FEES: {
+				type: 'object',
+				$ref: '#/fees',
+			},
+			MAX_PAYLOAD_LENGTH: {
+				type: 'integer',
+			},
+			MAX_PEERS: {
+				type: 'integer',
+			},
+			MAX_SHARED_TRANSACTIONS: {
+				type: 'integer',
+			},
+			MAX_TRANSACTIONS_PER_BLOCK: {
+				type: 'integer',
+			},
+			MAX_VOTES_PER_TRANSACTION: {
+				type: 'integer',
+			},
+			MAX_VOTES_PER_ACCOUNT: {
+				type: 'integer',
+			},
+			MIN_BROADHASH_CONSENSUS: {
+				type: 'integer',
+			},
+			MULTISIG_CONSTRAINTS: {
+				type: 'object',
+				required: ['MIN', 'LIFETIME', 'KEYSGROUP'],
+				properties: {
+					MIN: {
+						type: 'object',
+						required: ['MINIMUM', 'MAXIMUM'],
+						properties: {
+							MINIMUM: {
+								type: 'integer',
+							},
+							MAXIMUM: {
+								type: 'integer',
+							},
+						},
+					},
+					LIFETIME: {
+						type: 'object',
+						required: ['MINIMUM', 'MAXIMUM'],
+						properties: {
+							MINIMUM: {
+								type: 'integer',
+							},
+							MAXIMUM: {
+								type: 'integer',
+							},
+						},
+					},
+					KEYSGROUP: {
+						type: 'object',
+						required: ['MIN_ITEMS', 'MAX_ITEMS'],
+						properties: {
+							MIN_ITEMS: {
+								type: 'integer',
+							},
+							MAX_ITEMS: {
+								type: 'integer',
+							},
+						},
+					},
+				},
+			},
+			NETHASHES: {
+				type: 'array',
+				items: {
+					type: 'string',
+					pattern: '^[a-fA-F0-9]{64}$',
```

Comment:
We don't allow capital letters in any of hex strings. Please refer to `helpers/z_schema.formats.hex` for it. And wherever possible use the already defined formats.